File: android/Detonator/src/main/java/com/iconshot/detonator/request/TextAreaSetValueRequest.java
package com.iconshot.detonator.request;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.TextAreaElement;
import com.iconshot.detonator.renderer.Edge;

public class TextAreaSetValueRequest extends Request<String> {
    public TextAreaSetValueRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        String value = decodeData(String.class);

        Edge edge = getComponentEdge();

        TextAreaElement element = (TextAreaElement) edge.children.get(0).element;

        element.setValue(value);

        end();
    }
}

File: ios/Detonator/Sources/Detonator/Request/TextAreaSetValueRequest.swift
class TextAreaSetValueRequest: Request {
    override public func run() {
        let value: String! = decodeData()
        
        let edge = getComponentEdge()!
        
        let element = edge.children[0].element as! TextAreaElement
        
        element.setValue(value: value)
        
        end()
    }
}
